zaas-service/src/test/java/org/zowe/apiml/acceptance/ZaasTest.java

```diff
@@ -65,4 +73,41 @@ void givenZosmfCookieAndDummyAuthProvider_whenZoweJwtRequest_thenUnavailable() {
         //@formatter:on
     }
 
+    @Test
+    void givenNoCert_whenCallingSafCheck_then401() {
+
+        //@formatter:off
+        given().config(config().sslConfig(new SSLConfig().sslSocketFactory(
+                new SSLSocketFactory(httpConfig.getSecureSslContextWithoutKeystore(), ALLOW_ALL_HOSTNAME_VERIFIER)))
+            )
+            .contentType(APPLICATION_JSON)
+            .body(BODY)
+            .when()
+            .post(String.format("https://%s:%d/zaas/auth/check", hostname, port))
+            .then()
+            .statusCode(SC_UNAUTHORIZED);
+        //@formatter:on
+    }
+
+    @Test
+    void givenInvalidCert_whenCallingSafCheck_then401() {
+        SSLContext sslContext = httpConfig.secureSslContext();
+
+        RestAssuredConfig configWithCert = RestAssuredConfig.newConfig()
+            .sslConfig(new SSLConfig().sslSocketFactory(
```

Comment:
this one should be using a client cert that's actually mapped, the aim is to verify that a call to zaas should be made only with the gateway certificate first and the mapped client certificate in the forwarding headers

Reply:
You can use `SslContext` pre-configured ssl contexts for rest assured (it's in the apiml-common)

Reply:
thanks